include/z64hint_commands.h

```diff
@@ -0,0 +1,148 @@
+#ifndef Z64HINT_COMMANDS_H
+#define Z64HINT_COMMANDS_H
+
+#include "ultra64.h"
+
+/*
+ * Hint Command Types
+ */
+
+// Checks the condition and exits the script if the check passes
+#define HINT_CMD_TYPE_CHECK  0
+
+// Checks a chain of conditions and only evaluates the last entry
+// if all conditions leading up to it were also true
+#define HINT_CMD_TYPE_CHAIN  1
+
+// Checks a chain of conditionals and will return a randomly chosen entry among
+// the conditions that evaluated to true
+#define HINT_CMD_TYPE_RANDOM  2
+
+// Checks the condition and skips forward by specified number of commands if the check passes
+#define HINT_CMD_TYPE_SKIP   3
+
+// Always ends the script, returning the text id for this command
+#define HINT_CMD_TYPE_END    7
+
+/*
+ * Hint Condition Types
+ */
+
+// Check an eventChkInf flag
+#define HINT_CONDITION_FLAG          0
+
+// Check a dungeon item (map, compass, boss key)
+#define HINT_CONDITION_DUNGEON_ITEM  1
+
+// Check if an item is in an item slot
+#define HINT_CONDITION_ITEM          2
+
+// "Other" conditions described below
+#define HINT_CONDITION_OTHER         3
+
+/*
+ * "Other" Condition Types
+ */
+
+// Check what strength upgrade has been obtained so far
+#define HINT_CONDITION_STRENGTH_UPG  0
+
+// Check if specific boots have been obtained so far
+#define HINT_CONDITION_BOOTS         1
+
+// Check if a particular song has been obtained
+#define HINT_CONDITION_SONG          2
+
+// Check if a particular medallion has been obtained
+#define HINT_CONDITION_MEDALLION     3
+
+// Check if the magic meter has been obtained
+#define HINT_CONDITION_MAGIC         4
+
+/*
+ * Byte pack macros
+ */
+
+#define HINT_B0(type, condType, tf)            \
+    _SHIFTL(HINT_CMD_TYPE_##type, 5, 3) |      \
+    _SHIFTL(HINT_CONDITION_##condType, 1, 4) | \
+    _SHIFTL(tf, 0, 1)
+
+#define HINT_B1(condType, data)                \
+    _SHIFTL(HINT_CONDITION_##condType, 4, 4) | \
```

Comment:
I know that's how it was before, but this is an opportunity to not use `##` concatenation, as we've been avoiding it in order to facilitate finding things (e.g. with search-all)